### PR TITLE
docs: improve docs of assortativity

### LINKED ISSUE #1132: Replace  @param normalized TBD
in `assortativity()` docs, probably check the C docs

### PR DESCRIPTION
**Pull request by maelle**
Fix #1132 

Based on https://igraph.org/c/doc/igraph-Structural.html#igraph_assortativity

@szhorvat is this correct?

### CODE REVIEW

**Comment by szhorvat** [comment]
I'll need to have a more thorough read later, but the documentation should be read carefully and updated as a whole. We need to make sure that the description / formulas for assortativity match the updated included in the C core docs.

What happened here (in the C library) was that a more general version of assortativity for implemented, and the docs were significantly expanded.

**Comment by maelle** [comment]
I really hope we can find a solution for #1145 :smile_cat:

**Comment by szhorvat** [comment]
IMO that is not realistic, so let's not count on this happening automatically in the future. Even if it does happen, it will take many many months.

**Comment by szhorvat** [comment]
Can you fix the top paragraph?

> The assortativity coefficient is positive **if** similar

Note the word I bolded, `s/is/if`

**Comment by szhorvat** [comment]
Please add this somewhere at the bottom:

> Undirected graphs are effectively treated as directed ones with all-reciprocal edges. Thus, self-loops are taken into account twice in undirected graphs.

**Comment by szhorvat** [comment]
There are some `sigma` which should be `\sigma` in LaTeX notation. 

image

**Comment by szhorvat** [comment]
Please remove `(minus one)` from:

> assortativity_degree() uses vertex degree (minus one) as vertex values and calls assortativity().

In the end it makes no difference if one is subtracted or not.

File: R/assortativity.R [hunk 1]
```diff
@@ -116,7 +116,8 @@ assortativity.degree <- function(graph, directed = TRUE) { # nocov start
 #'   `TRUE` here to do the natural thing, i.e. use directed version of the
 #'   measure for directed graphs and the undirected version for undirected
 #'   graphs.
-#' @param normalized TBD
+#' @param normalized Boolean, whether to compute the normalized covariance,
+#'  i.e. Pearson correlation. Supply true here to compute the standard assortativity.
```
Comment (szhorvat) [inline R/assortativity.R:120]:
```suggestion
#' @param normalized Boolean, whether to compute the normalized assortativity.
#' The non-normalized nominal assortativity is identical to modularity.
#' The non-normalized value-based assortativity is simply the covariance of the
#' values at the two ends of edges.
```

We should later elaborate on what "normalized" means separately for `assortativity()` and `assortativity_nominal()`. The latter is not a Pearson correlation.

**Comment by szhorvat** [comment]
I'm sorry, this was a very sloppy review by me, but with this the docs should at least be accurate and complete. We can fix them up when we have the time, after 2.0 ...

**Comment by maelle** [comment]
It didn't seem sloppy! I've made changes.

**Comment by maelle** [comment]
@szhorvat could you please re-review this? :pray:

**Review by szhorvat — APPROVED** [review]
I think this should be merged now, as it is all correct. I'll try to make improvements later. Thank you Maëlle!

**Pull request merged**